discovery-provider/src/api/v1/users.py

```diff
@@ -947,3 +954,74 @@ def get(self, id: str):
         support = get_support_sent_by_user(args)
         support = list(map(extend_supporting, support))
         return success_response(support)
+
+
+verify_token_response = make_response(
+    "verify_token", ns, fields.List(fields.Nested(decoded_user_token))
+)
+
+
+@ns.route("/verify_token")
+class GetTokenVerification(Resource):
+    @record_metrics
+    @ns.doc(
+        id="""Verify ID token""",
+        description="""Verify if the given jwt ID token was signed by the subject (user) in the payload""",
+        responses={
+            200: "Success",
+            400: "Bad input",
+            404: "ID token not valid",
+            500: "Server error",
+        },
+    )
+    @ns.expect(verify_token_parser)
+    @ns.marshal_with(verify_token_response)
+    def get(self):
+        args = verify_token_parser.parse_args()
+        # 1. Break JWT into parts
+        raw_token = args["token"]
+        if not raw_token:
+            abort_bad_request_param("token", ns)
+        token_parts = args["token"].split(".")
+        if not len(token_parts) == 3:
+            abort_bad_request_param("token", ns)
+
+        # 2. Decode the signature
+        try:
+            signature = base64.urlsafe_b64decode(token_parts[2] + "==")
+        except Exception:
+            ns.abort(400, "JWT signature could not be decoded.")
+
+        signature = signature.decode()
+        base64_header = token_parts[0]
+        base64_payload = token_parts[1]
+        message = f"{base64_header}.{base64_payload}"
+
+        # 3. Recover message from signature
+        web3 = web3_provider.get_web3()
+
+        wallet = None
+        encoded_message = encode_defunct(text=message)
+        try:
+            wallet = web3.eth.account.recover_message(
+                encoded_message,
+                signature=signature,
+            )
+        except Exception:
+            ns.abort(404, "The JWT signature is invalid.")
+        if not wallet:
+            ns.abort(404, "The JWT signature is invalid.")
```

Comment:
nit, the same error message is used for all the ns.aborts, could be more specific for each to differentiate when it errored